llama_index/node_parser/extractors/marvin_entity_extractor.py

```diff
@@ -0,0 +1,88 @@
+from typing import TYPE_CHECKING, Any, List, Optional, Sequence, Type, Dict, cast
+
+if TYPE_CHECKING:
+    from marvin import AIModel
+
+from pydantic import BaseModel, Field
+
+from llama_index.schema import BaseNode, TextNode
+
+from llama_index.node_parser.extractors.metadata_extractors import (
+    MetadataFeatureExtractor,
+)
+
+
+class MarvinEntityExtractor(MetadataFeatureExtractor):
+    # Forward reference to handle circular imports
+    marvin_model: Type["AIModel"] = Field(
+        description="The Marvin model to use for extracting entities"
+    )
+    llm_model_string: Optional[str] = Field(
+        description="The LLM model string to use for extracting entities"
+    )
+
+    """Entity extractor for cusstom entities using Marvin. 
+    Node-level extractor. Extracts
+    `marvin_entities` metadata field.
+    Args:
+        marvin_model: Marvin model to use for extracting entities
+        llm_model_string: (optional) LLM model string to use for extracting entities
```

Comment:
So just to confirm, this is using an LLM to extract entities?

Reply:
(Side note, running the extraction took 10 minutes, and I hit a "bad gateway, Error 502" before it finished)

Reply:
Seems like the extracted "entities" aren't really entities, but they do seem very useful!

Reply:
I actually use it populate custom entities ... I tried to come up with a good example.  An entity is any domain object important to your domain.

Reply:
I had it run in about 4 minutes using gpt-3.5-turbo ... but with gpt-4 it took 10 minutes.  yes delete some nodes if you need to

Reply:
Yes ... it's using an llm to extract custom entities or to extract whatever you want really.  You just define the model in Marvin.  Maybe the name should just be changed to MarvinMetaDataExtractor because you can really extract whatever you want

Reply:
You can also have a hierarchical model using an ai_model as the top  level and normal pydantic models for the properties or strings or ints etc

Reply:
Actually using it for a production use case and it's very powerful for our needs.  This is a very simplified version but I believe it captures the essence.

Reply:
It also works well with Claude ... I'll stop spamming now ... lol

Reply:
Sorry if it sounded like I was critiquing it 😅  cool to know the use cases for it! Was just very inexperienced with what it did 🫡👍

Reply:
Critiquing is good ... I have thick skin

Reply:
I can take a look in the morning ... no problem ...